Lib/Common/RetryPolicies/ExponentialRetry.cs

```diff
@@ -97,6 +98,14 @@ public bool ShouldRetry(int currentRetryCount, int statusCode, Exception lastExc
             return false;
         }
 
+        private double CalculateIncrement(int currentRetryCount)
+        {
+            lock (randomLock)
```

Comment:
Why not use

```
public static class StaticRandom
{
    static int seed = Environment.TickCount;

    static readonly ThreadLocal<Random> random =
        new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref seed)));

    public static int Rand()
    {
        return random.Value.Next();
    }
}
```

https://stackoverflow.com/questions/19270507/correct-way-to-use-random-in-multithread-application

to avoid lock?